lib/msal-common/src/cache/entities/AppMetadataEntity.ts

```diff
@@ -27,7 +27,19 @@ export class AppMetadataEntity {
      * Generate Account Cache Key as per the schema: <home_account_id>-<environment>-<realm*>
      */
     generateAppMetaDataEntityKey(): string {
-        const appMetaDataKeyArray: Array<string> = [APP_META_DATA, this.environment, this.clientId];
+        const appMetaDataKeyArray: Array<string> = [APP_METADATA, this.environment, this.clientId];
         return appMetaDataKeyArray.join(Separators.CACHE_KEY_SEPARATOR).toLowerCase();
     }
+
+    /**
+     * Validates an entity: checks for all expected params
+     * @param entity
+     */
+    static isAppMetadataEntity(key: string, entity: object): boolean {
```

Comment:
Whats the difference between this function and `isAppMetadata` in `CacheManager`?

Reply:
We are also checking the entity.

Reply:
Curious what the scenarios are for having both functions? Would `isAppMetadataEntity` be enough to cover all scenarios?

Reply:
There may not be. These are the validations we are providing and why:
* **Key based validation** - applicable in `getByFilter` APIs in `CacheManager`, we should be able to fetch the key and determine if we want to fetch the value; it is performance degrading to fetch the value for all keys in a loop. This is currently supported by `getCredentialType()` and `isAppMetadata()` in `CacheManager` today. Moving them to `entities` as `static APIs` may make more sense now.

* **Value based validation** - This is what we are adding now. This comes to play when we are in `getItem()` fetching either the instance(`msal-node`) or a JSON object(`msal-browser`), and verify the value that we are fetching is indeed what we want.

Reply:
Got it, that makes sense. Thanks!